feat: enable support to Blockaid on custom networks on mobile

blackdevelopa opened the pull request:
## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This is a follow up to https://github.com/MetaMask/metamask-mobile/pull/7547. Once this PR is merged, we'll be able to support additional networks on extension and we should enable the feature on them.

We should enabled it on the following networks:

Ethereum Mainnet (already done)
BNB chain
Polygon
Arbitrum
Optimism
Avalanche
Linea

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1693

## **Manual testing steps**

1. Go to Settings >> Network
2. Add custom networks
3. Go to Settings >> Experimental
4. See that blockaid toggle is only enabled for supported networks

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

http://recordit.co/jC95wVGXEi



https://github.com/MetaMask/metamask-mobile/assets/29962968/dfcb4d8e-edb1-4698-8d6f-e1a44c15597e




## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

blackdevelopa committed:
blockaid support for more networks

blackdevelopa committed:
revert package change

blackdevelopa committed:
update ppom validator

blackdevelopa committed:
updates

jpuri committed:
Update PPOM validator (#8275)

## **Description**

Fix blockaid mobile multichain. Re-init PPOM instance on network switch.

## **Related issues**

Fixes: MetaMask/MetaMask-planning#1926

## **Manual testing steps**

1. Open app and enable blockaid
2. Switch to mainnet, wait for few seconds and go to test dapp to submit
malicious transaction - validation should work
3. Switch to any other network supported by blockaid and wait for few
seconds and go to test dapp to submit malicious transaction - validation
should work

## **Screenshots/Recordings**
NA

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it
is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [X] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>

jpuri commented on app/components/UI/BlockaidBanner/BlockaidBanner.tsx:
```diff
@@ -101,7 +101,7 @@ const BlockaidBanner = (bannerProps: BlockaidBannerProps) => {
     !isBlockaidFeatureEnabled() ||
-    !isMainnetByChainId(chainId) ||
+    !SUPPORTED_CHAIN_IDS.includes(chainId) ||
     !isSecurityAlertsEnabled
```
@blackdevelopa : it will be useful to test cover above condition.

jpuri commented on app/lib/ppom/ppom-util.ts:
```diff
@@ -46,7 +46,7 @@ const validateRequest = async (req: any, transactionId?: string) => {
-    !isMainnetByChainId(currentChainId)
+    !SUPPORTED_CHAIN_IDS.includes(currentChainId)
   ) {
     return;
```
@blackdevelopa : it will be useful to test cover above condition.

jpuri commented on app/util/networks/index.js:
```diff
@@ -110,6 +110,16 @@ const NetworkList = {
+  NETWORKS_CHAIN_ID.OPTIMISM,
+  NETWORKS_CHAIN_ID.AVAXCCHAIN,
+  NETWORKS_CHAIN_ID.LINEA_MAINNET,
+];
```
May be rename to `BLOCKAID_ SUPPORTED_CHAIN_IDS`

jpuri committed:
Blockaid mobile performance improvements (#8343)

blackdevelopa committed:
feedback

blackdevelopa committed:
snapshot

jpuri commented:
@SocketSecurity ignore @metamask/ppom-validator@0.22.0
@SocketSecurity ignore react-native-quick-crypto@0.6.1

jpuri commented:
@SocketSecurity ignore npm/react-native-quick-crypto@0.6.1
@SocketSecurity ignore npm/@metamask/ppom-validator@0.22.0

blackdevelopa committed:
small fix

blackdevelopa committed:
small change

jpuri committed:
Using existing setup file testSetup.js for mocking (#8351)

blackdevelopa committed:
fix: Update what's new and settings copy to communicate additional network coverage on mobile (#8327)

## **Description**

This is a follow up to
https://github.com/MetaMask/MetaMask-planning/issues/1693. Once that PR
is merged, we should adjust copy to communicate the support for custom
networks.

What's new copy should be updated to:
Steer clear of known scams while still preserving your privacy with
security alerts powered by Blockaid. This feature is available on
Arbitrum, Avalanche, BNB chain, Ethereum Mainnet, Linea, Optimism and
Polygon.

Always do your own due diligence before approving requests.

Settings copy should be updated to:
Privacy preserving - no data is shared with third parties. Available on
Arbitrum, Avalanche, BNB chain, Ethereum Mainnet, Optimism and Polygon.

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

blackdevelopa committed:
rebased to main

blackdevelopa committed:
removed unused file

blackdevelopa committed:
build fail fix

jpuri committed:
Fix display of blockaid banner

jpuri committed:
Fix display of blockaid banner

jpuri committed:
fix build

jpuri committed:
Merge branch 'main' into 1963/blockaid_custom_network

blackdevelopa committed:
Merge 125da6958df83b86eeae64293b3f4a310d2bee92 into b8741c7b3310165aebab8e0605da1240fad2269d

metamaskbot committed:
Bump commit for Bitrise

blackdevelopa committed:
Merge branch 'main' into 1963/blockaid_custom_network

jpuri committed:
update

jpuri committed:
Merge branch 'check_qa_build' into 1963/blockaid_custom_network

jpuri committed:
Revert "update"

This reverts commit 067f91001079da05dc2b47d13a28a8b2c8d27463.

jpuri committed:
Update

jpuri committed:
Update

jpuri committed:
Update

jpuri committed:
Update

jpuri committed:
Update

jpuri committed:
update
